docs/reference/modules/gateway.asciidoc

```diff
@@ -56,9 +56,29 @@ NOTE: These settings only take effect on a full cluster restart.
 [[dangling-indices]]
 ==== Dangling indices
 
-When a node joins the cluster, any shards stored in its local data
-directory which do not already exist in the cluster will be imported into the
-cluster. This functionality is intended as a best effort to help users who
-lose all master nodes. If a new master node is started which is unaware of
-the other indices in the cluster, adding the old nodes will cause the old
-indices to be imported, instead of being deleted.
+When a node joins the cluster, if it finds any shards stored in its local data
+directory that do not already exist in the cluster, it will consider those
+shards to be "dangling". Importing dangling indices
+into the cluster using `gateway.auto_import_dangling_indices` is not safe.
+Instead, use the <<dangling-indices-api,Dangling indices API>>. Neither
+mechanism provides any guarantees as to whether the imported data truly
+represents the latest state of the data when the index was still part of
+the cluster.
+
+`gateway.auto_import_dangling_indices`::
+
+    deprecated:[7.9.0, This setting will be removed in 8.0. You should use the dedicated dangling indices API instead.]
+    Whether to automatically import dangling indices into the cluster
+    state, provided no indices already exist with the same name. Defaults
+    to `false`.
+
+WARNING: The auto-import functionality was intended as a best effort to help users
+who lose all master nodes. For example, if a new master node were to be
+started which was unaware of the other indices in the cluster, adding the
+old nodes would cause the old indices to be imported, instead of being
+deleted. However there are several issues with automatic importing, and
+its use is strongly discouraged in favour of the
+<<dangling-indices-api,dedicated API>.
+
+WARNING: Losing all master nodes is a situation that should be avoided at
```

Comment:
@lockewritesdocs Is it overkill to have two `WARNING:` sections? I felt that they were making separate points, and all the content here ought to be scary.